### ISSUE TITLE
about predscore error

### ISSUE THREAD

**Issue by JangMyoungjin** [cma2015/DeepGS#13]
Hello,
THank you for the package. I am getting an error. I am getting same prediction score for all the tested samples. I am attaching the code i used and the files that i used (100 samples, 200 markers and 1 phenotype).  I had input the marker file as matrix and phenotype file as named vector. Can you please help.

>hanwoo <- read.table("hanwoo_100_200.raw",header = T)
>pheno <- as.vector(hanwoo$PHENOTYPE)
>names(pheno) <- rownames(hanwoo)
>markers<- hanwoo[,-1]

>y <- pheno
>Markers <- markers

>cvSampleList <- cvSampleIndex(length(y),10,1)
>cvIdx <- 1

>trainIdx <- cvSampleList[[cvIdx]]$trainIdx
>testIdx <- cvSampleList[[cvIdx]]$testIdx
>trainMat = Markers[trainIdx,]
>trainPheno = y[trainIdx]
>validIdx <- sample(1:length(trainIdx),floor(length(trainIdx)*0.1))
>validMat <- trainMat[validIdx,]
>validPheno <- trainPheno[validIdx]
>testMat = Markers[testIdx,]
>testPheno = y[testIdx]

>conv_kernel <- c("1*18") # convolution kernels (fileter shape)
>conv_stride <- c("1*1")
>conv_num_filter <- c(8)  # number of filters
>pool_act_type <- c("relu") # active function for next pool
>pool_type <- c("max") # max pooling shape
>pool_kernel <- c("1*4") # pooling shape
>pool_stride <- c("1*4") # number of pool kernerls
>fullayer_num_hidden <- c(32,1)
>fullayer_act_type <- c("sigmoid")
>drop_float <- c(0.2,0.1,0.05)
>cnnFrame <- list(conv_kernel =conv_kernel,conv_num_filter = conv_num_filter,
                 conv_stride = conv_stride,pool_act_type = pool_act_type,
                 pool_type = pool_type,pool_kernel =pool_kernel,
                 pool_stride = pool_stride,fullayer_num_hidden= fullayer_num_hidden,
                 fullayer_act_type = fullayer_act_type,drop_float = drop_float)

>markerImage = paste0("1*",ncol(trainMat))

 >DeepGS_obj <- train_deepGSModel(trainMat = trainMat,trainPheno = trainPheno,
                                 validMat = validMat,validPheno = validPheno, markerImage = markerImage,
                                 cnnFrame = cnnFrame,device_type = "cpu",gpuNum = 1, eval_metric = "mae",
                                 num_round = 6000,array_batch_size= 5,learning_rate = 0.01,
                                 momentum = 0.5,wd = 0.00001, randomseeds = 0,initializer_idx = 0.01,
                                 verbose =TRUE)

> DeepGS_pred <- predict_GSModel(GSModel = DeepGS_obj,testMat = Markers[testIdx,],
                                markerImage = markerImage )

>write.csv(y, "Hanwoo_pheno.txt")
>write.csv(Markers, "Hanwoo_marker.txt")

[deepgs.zip](https://github.com/cma2015/DeepGS/files/5259545/deepgs.zip)

**Comment by zhixuqiu** [comment]
> Hello,
> THank you for the package. I am getting an error. I am getting same prediction score for all the tested samples. I am attaching the code i used and the files that i used (100 samples, 200 markers and 1 phenotype). I had input the marker file as matrix and phenotype file as named vector. Can you please help.
> 
> > hanwoo <- read.table("hanwoo_100_200.raw",header = T)
> > pheno <- as.vector(hanwoo$PHENOTYPE)
> > names(pheno) <- rownames(hanwoo)
> > markers<- hanwoo[,-1]
> 
> > y <- pheno
> > Markers <- markers
> 
> > cvSampleList <- cvSampleIndex(length(y),10,1)
> > cvIdx <- 1
> 
> > trainIdx <- cvSampleList[[cvIdx]]$trainIdx
> > testIdx <- cvSampleList[[cvIdx]]$testIdx
> > trainMat = Markers[trainIdx,]
> > trainPheno = y[trainIdx]
> > validIdx <- sample(1:length(trainIdx),floor(length(trainIdx)*0.1))
> > validMat <- trainMat[validIdx,]
> > validPheno <- trainPheno[validIdx]
> > testMat = Markers[testIdx,]
> > testPheno = y[testIdx]
> 
> > conv_kernel <- c("1_18") # convolution kernels (fileter shape)
> > conv_stride <- c("1_1")
> > conv_num_filter <- c(8)  # number of filters
> > pool_act_type <- c("relu") # active function for next pool
> > pool_type <- c("max") # max pooling shape
> > pool_kernel <- c("1_4") # pooling shape
> > pool_stride <- c("1_4") # number of pool kernerls
> > fullayer_num_hidden <- c(32,1)
> > fullayer_act_type <- c("sigmoid")
> > drop_float <- c(0.2,0.1,0.05)
> > cnnFrame <- list(conv_kernel =conv_kernel,conv_num_filter = conv_num_filter,
> > conv_stride = conv_stride,pool_act_type = pool_act_type,
> > pool_type = pool_type,pool_kernel =pool_kernel,
> > pool_stride = pool_stride,fullayer_num_hidden= fullayer_num_hidden,
> > fullayer_act_type = fullayer_act_type,drop_float = drop_float)
> 
> > markerImage = paste0("1*",ncol(trainMat))
> 
> > DeepGS_obj <- train_deepGSModel(trainMat = trainMat,trainPheno = trainPheno,
> > validMat = validMat,validPheno = validPheno, markerImage = markerImage,
> > cnnFrame = cnnFrame,device_type = "cpu",gpuNum = 1, eval_metric = "mae",
> > num_round = 6000,array_batch_size= 5,learning_rate = 0.01,
> > momentum = 0.5,wd = 0.00001, randomseeds = 0,initializer_idx = 0.01,
> > verbose =TRUE)
> 
> > DeepGS_pred <- predict_GSModel(GSModel = DeepGS_obj,testMat = Markers[testIdx,],
> > markerImage = markerImage )
> 
> > write.csv(y, "Hanwoo_pheno.txt")
> > write.csv(Markers, "Hanwoo_marker.txt")
> 
> [deepgs.zip](https://github.com/cma2015/DeepGS/files/5259545/deepgs.zip)

**Comment by zhixuqiu** [comment]
> Hello,
> THank you for the package. I am getting an error. I am getting same prediction score for all the tested samples. I am attaching the code i used and the files that i used (100 samples, 200 markers and 1 phenotype). I had input the marker file as matrix and phenotype file as named vector. Can you please help.
> 
> > hanwoo <- read.table("hanwoo_100_200.raw",header = T)
> > pheno <- as.vector(hanwoo$PHENOTYPE)
> > names(pheno) <- rownames(hanwoo)
> > markers<- hanwoo[,-1]
> 
> > y <- pheno
> > Markers <- markers
> 
> > cvSampleList <- cvSampleIndex(length(y),10,1)
> > cvIdx <- 1
> 
> > trainIdx <- cvSampleList[[cvIdx]]$trainIdx
> > testIdx <- cvSampleList[[cvIdx]]$testIdx
> > trainMat = Markers[trainIdx,]
> > trainPheno = y[trainIdx]
> > validIdx <- sample(1:length(trainIdx),floor(length(trainIdx)*0.1))
> > validMat <- trainMat[validIdx,]
> > validPheno <- trainPheno[validIdx]
> > testMat = Markers[testIdx,]
> > testPheno = y[testIdx]
> 
> > conv_kernel <- c("1_18") # convolution kernels (fileter shape)
> > conv_stride <- c("1_1")
> > conv_num_filter <- c(8)  # number of filters
> > pool_act_type <- c("relu") # active function for next pool
> > pool_type <- c("max") # max pooling shape
> > pool_kernel <- c("1_4") # pooling shape
> > pool_stride <- c("1_4") # number of pool kernerls
> > fullayer_num_hidden <- c(32,1)
> > fullayer_act_type <- c("sigmoid")
> > drop_float <- c(0.2,0.1,0.05)
> > cnnFrame <- list(conv_kernel =conv_kernel,conv_num_filter = conv_num_filter,
> > conv_stride = conv_stride,pool_act_type = pool_act_type,
> > pool_type = pool_type,pool_kernel =pool_kernel,
> > pool_stride = pool_stride,fullayer_num_hidden= fullayer_num_hidden,
> > fullayer_act_type = fullayer_act_type,drop_float = drop_float)
> 
> > markerImage = paste0("1*",ncol(trainMat))
> 
> > DeepGS_obj <- train_deepGSModel(trainMat = trainMat,trainPheno = trainPheno,
> > validMat = validMat,validPheno = validPheno, markerImage = markerImage,
> > cnnFrame = cnnFrame,device_type = "cpu",gpuNum = 1, eval_metric = "mae",
> > num_round = 6000,array_batch_size= 5,learning_rate = 0.01,
> > momentum = 0.5,wd = 0.00001, randomseeds = 0,initializer_idx = 0.01,
> > verbose =TRUE)
> 
> > DeepGS_pred <- predict_GSModel(GSModel = DeepGS_obj,testMat = Markers[testIdx,],
> > markerImage = markerImage )
> 
> > write.csv(y, "Hanwoo_pheno.txt")
> > write.csv(Markers, "Hanwoo_marker.txt")
> 
> [deepgs.zip](https://github.com/cma2015/DeepGS/files/5259545/deepgs.zip)

Thank you for your interest in DeepGS. There are some tips for using DeepGS. First, the genotype matrix doesn't support input NA values. Second, you can standardize your phenotype values when the phenotype has large range, such as z score method.
For example, You can add commands as shown below in the code.
`y <- pheno
Markers <- markers
y <- scale(y)
Markers <- as.matrix(Markers)
Markers[which(is.na(Markers))] <- 0
`
